frame/grandpa/src/lib.rs

```diff
@@ -638,14 +626,26 @@ impl<T: Trait> pallet_session::OneSessionHandler<T::AccountId> for Module<T>
 		// Always issue a change if `session` says that the validators have changed.
 		// Even if their session keys are the same as before, the underlying economic
 		// identities have changed.
-		let current_set_id = if changed {
+		let current_set_id = if changed || <Stalled<T>>::exists() {
```

Comment:
Even when the session module reports no changes, if there is a stall we want to schedule a forced change explicitly (regardless of whether it is the same authorities as in the last session).